test/parallel/test-child-process-argument-errors.js

```diff
@@ -0,0 +1,275 @@
+  //   (f, o)
+  //   (f, o, c)
+  //   (f, c)
+  execFile(cmd);
+  execFile(cmd, a);
+  execFile(cmd, a, o);
+  execFile(cmd, a, o, c);
+  execFile(cmd, a, c);
+  execFile(cmd, o);
+  execFile(cmd, o, c);
+  execFile(cmd, c);
+
+  // Variants of undefined as explicit 'no argument' at a position.
+  execFile(cmd, u, o, c);
+  execFile(cmd, a, u, c);
+  execFile(cmd, a, o, u);
+  execFile(cmd, n, o, c);
+  execFile(cmd, a, n, c);
+  execFile(cmd, a, o, n);
+  execFile(cmd, u, u, u);
+  execFile(cmd, u, u, c);
+  execFile(cmd, u, o, u);
+  execFile(cmd, a, u, u);
+  execFile(cmd, n, n, n);
+  execFile(cmd, n, n, c);
+  execFile(cmd, n, o, n);
+  execFile(cmd, a, n, n);
+  execFile(cmd, a, u);
+  execFile(cmd, a, n);
+  execFile(cmd, o, u);
+  execFile(cmd, o, n);
+  execFile(cmd, c, u);
+  execFile(cmd, c, n);
+  execFile(cmd, c, s);
+}
+
+// Verify that invalid argument combinations throw.
+{
+  const errorStart =
+    'Function was invoked with an unknown combination of arguments.\n' +
+    'Expected child_process.execFile(file[, args][, options][, callback])\n';
+
+  // String is invalid in arg position (this may seem strange, but is
+  // consistent across node API, cf. `net.createServer('not options', 'not
+  // callback')`
+  invalidSignatureError(
+    () => execFile(cmd, s, o, c),
+    errorStart +
+      `Received child_process.execFile('${cmd}', 'string', {}, [Function: c])`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, a, s, c),
+    errorStart +
+    `Received child_process.execFile('${cmd}', [], 'string', [Function: c])`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, a, o, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', [], {}, 'string')`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, a, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', [], 'string')`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, o, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', {}, 'string')`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, u, u, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', undefined, undefined, 'string')`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, n, n, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', null, null, 'string')`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, a, u, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', [], undefined, 'string')`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, a, n, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', [], null, 'string')`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, u, o, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', undefined, {}, 'string')`
+  );
+
+  invalidSignatureError(
+    () => execFile(cmd, n, o, s),
+    errorStart +
+    `Received child_process.execFile('${cmd}', null, {}, 'string')`
+  );
```

Comment:
I personally prefer to avoid loops and argument manipulations for this kind of tests because if any of them fails, it's hard to see what's going on from the first glance if you are not familiar with the test (or even for the author themselves after a sufficiently long time..I've been there).

Reply:
That is why I normally add a special error message in those cases that shows the input values and the test number that ran.